### PR TITLE
Squashed commit of "Add Column Metadata #152"

### PR DESCRIPTION
**Pull request by ephemeralowl**
Work done by MarcelHillmann@ to add column metadata, but that pull request hasn't seen activity in some time.

### CODE REVIEW

**Comment by l3pp4rd** [comment]
hi, sorry, very busy these months, but I need to find time to closely review it

**Comment by jopicornell** [comment]
Hi there!

I'm using [go-jet](https://github.com/go-jet/jet) to build my SQL for a little project, and the issue that tries to fix this PR stops me to write tests for my project. Seems like go-jet is taking the ColumnTypes of rows, but databaseType is an empty string and it's taking all as string, failing to use their qrm as a struct scanner. 

I'm pointing this out here so that everyone who wants to use go-jet's qrm (it's like an orm, but for building quieries) and use this library, this issue will stop them to write tests.

If you need anything in this PR I'll help you with anything I can.

File: rows.go [hunk 1]
```diff
@@ -115,24 +151,63 @@ func (rs *rowSets) invalidateRaw() {
 	rs.raw = nil
 }
 
+// Column is a mocked column Metadata for rows.ColumnTypes()
+type Column struct {
+	name, dbTyp              string
```
Comment (l3pp4rd) [inline rows.go]: can we make all these fields exported, removing `NewColumn` and `NewColumnSimple` functions. and move everything related to Column in `column.go` file

Comment (l3pp4rd) [reply rows.go]: @jopicornell these are the only changes I would like to be made here

**Pull request closed without merging**